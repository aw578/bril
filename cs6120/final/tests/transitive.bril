@main {
  x: int = const 0;
  y: int = const 1;
  z: int = const 2;
  w: int = const 3;
  a: int = add x y;
  b: int = add x y;
  d: int = add a z;
  e: int = add b z;
  f: int = add d w;
  g: int = add e w;
}