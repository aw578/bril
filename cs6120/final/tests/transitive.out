Congruence classes for function main:
Class const 0: ['x.0']
Class const 1: ['y.0']
Class const 2: ['z.0']
Class const 3: ['w.0']
Class add: ['a.0', 'b.0']
Class add_0: ['f.0', 'g.0']
Class add_0_0: ['d.0', 'e.0']
